Documentation/ja/TestUsingMock.md

```diff
@@ -0,0 +1,153 @@
+# モックを使ったテスト
+
+## モックとは
+
+モックとはテスト対象のオブジェクト(クラス、構造体)が呼び出し先のオブジェクトと意図したとおりに協調動作するかどうかをテストするために使うオブジェクトのことです。
+
+## Swift でモックを使ったテストを書く
+
+### サンプルアプリケーション	
+
+ここでは例として RSS reader のような記事の一覧を UITableView　で表示アプリケーションを考えます。
+
+* Article という構造体(struct)を ArticleViewController の UITableView で表示する
+* データ(Article)の取得や保持、UITableViewDataSource の処理は ArticleProviderProtocol を実装したクラスが行う
+
+ここで ArticleProviderProtocol を定義します。
+
+```swift
+// Swift
+protocol ArticleProviderProtocol: UITableViewDataSource {
+    var articles: [Article] { get }
+    weak var tableView: UITableView! { get set }
+    func setup()
+    func fetch()
+}
+```
```

Comment:
Would it be possible to demonstrate a problem that can be solved with mocks *without* using UIKit? I'm concerned about readers who may not be 100% familiar with UIKit and `UITableViewDataSource`. Just wondering whether that would make this easier to read--thoughts? :raised_hands: